lib/src/builders/class_builder.dart

```diff
@@ -5,79 +5,96 @@
 part of code_builder;
 
 /// Builds a [ClassDeclaration] AST.
-class ClassBuilder extends _AbstractCodeBuilder<ClassDeclaration> {
+class ClassBuilder implements CodeBuilder<ClassDeclaration> {
   static Token _abstract = new KeywordToken(Keyword.ABSTRACT, 0);
   static Token _extends = new KeywordToken(Keyword.EXTENDS, 0);
   static Token _implements = new KeywordToken(Keyword.IMPLEMENTS, 0);
   static Token _with = new KeywordToken(Keyword.WITH, 0);
 
+  final String _name;
+  final bool _isAbstract;
+  final TypeBuilder _extend;
+  final Iterable<TypeBuilder> _implement;
+  final Iterable<TypeBuilder> _mixin;
+
+  final List<FieldBuilder> _fields = <FieldBuilder>[];
+  final List<MethodBuilder> _methods = <MethodBuilder>[];
+  final List<AnnotationBuilder> _metadata = <AnnotationBuilder>[];
+  final List<ConstructorBuilder> _constructors = <ConstructorBuilder>[];
+
   /// Create a new builder for a `class` named [name].
   ///
   /// Optionally, define another class to [extend] or classes to either
   /// [implement] or [mixin]. You may also define a `class` as [abstract].
   factory ClassBuilder(
     String name, {
     bool abstract: false,
-    String extend,
-    Iterable<String> implement: const [],
-    Iterable<String> mixin: const [],
-  }) {
-    var astNode = _emptyClassDeclaration()..name = _stringId(name);
-    if (abstract) {
-      astNode.abstractKeyword = _abstract;
-    }
-    if (extend != null) {
-      astNode.extendsClause = new ExtendsClause(
-          _extends,
-          new TypeName(
-            _stringId(extend),
-            null,
-          ));
-    }
-    if (implement.isNotEmpty) {
-      astNode.implementsClause = new ImplementsClause(
-          _implements,
-          implement
-              .map/*<TypeName>*/((i) => new TypeName(_stringId(i), null))
-              .toList());
-    }
-    if (mixin.isNotEmpty) {
-      astNode.withClause = new WithClause(
-          _with,
-          mixin
-              .map/*<TypeName>*/((i) => new TypeName(_stringId(i), null))
-              .toList());
-    }
-    return new ClassBuilder._(astNode);
-  }
+    TypeBuilder extend,
+    Iterable<TypeBuilder> implement: const [],
+    Iterable<TypeBuilder> mixin: const [],
+  }) =>
+      new ClassBuilder._(
+        name,
+        abstract,
```

Comment:
While legal Dart code I'm not a fan of Dart keywords used as identifiers, especially in the public API (this named parameter is visible to user code)